packages/yoastseo/src/languageProcessing/researches/wordComplexity.js

```diff
@@ -1,51 +1,118 @@
 import getWords from "../helpers/word/getWords.js";
-import countSyllables from "../helpers/syllables/countSyllables.js";
 import getSentences from "../helpers/sentence/getSentences.js";
+import { flatMap } from "lodash-es";
+
+/**
+ * Checks if a word is complex.
+ *
+ * @param {string} word     The word to check.
+ * @param {object} config   The config to use.
+ *
+ * @returns {boolean}    Whether or not a word is complex.
+ */
+const checkIfWordIsComplex = function( word, config ) {
+	const lengthLimit = config.wordLength;
+	const frequencyList = config.frequencyList;
+	// Whether uppercased beginning of a word decreases its complexity.
+	const doesUpperCaseDecreaseComplexity = config.doesUpperCaseDecreasesComplexity;
+
+	let isWordComplex = false;
+	if ( frequencyList.length === 0 ) {
+		return;
+	}
+	/*
+	 * Check for each word whether it is a complex word or not.
+	 * A word is complex if:
+	 * its length is longer than the limit, AND
+	 * the word is not in the frequency list, AND
+	 * if the word does NOT start with a capital letter
+	 * (for languages that see long words to be less complex if they start with a capital letter)
+	 */
+	if ( word.length > lengthLimit ) {
+		if ( ! frequencyList.includes( word ) ) {
+			if ( doesUpperCaseDecreaseComplexity === true && word[ 0 ].toLowerCase() === word[ 0 ] ) {
+				isWordComplex = true;
+			}
+		}
+	}
```

Comment:
The third check still needs to be separate, since other languages might have `doesUpperCaseDecreaseComplexity` to be false. And in that case, we need to add a different check anyway. However, I will combine the first two checks with `&&`